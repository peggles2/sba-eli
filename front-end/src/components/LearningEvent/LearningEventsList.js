import React, { Component } from "react";
import { List } from "semantic-ui-react";
import { connect } from "react-redux";
import LearningEventsItem from "./LearningEventsItem";
import { getLearningEvents } from "../../actions/learningEventActions";

export class LearningEventsList extends Component {
  componentDidMount() {
    this.eventsList();
  }

  eventsList() {
    this.props.dispatch(getLearningEvents(this.props.course_id, this.props.module_id));
  }

  renderEventsList(events = []) {
    if (events.length) {
      return events.map(event => {
        return <LearningEventsItem key={event.id} item={event} />;
      });
    }
    return <li>No data</li>;
  }

  render() {
    return <List>{this.renderEventsList(this.props.learningEvents)}</List>;
  }
}

const mapStateToProps = store => {
  return {
    learningEvents: store.learningEvent.learningEvents
  };
};

export default connect(mapStateToProps)(LearningEventsList)