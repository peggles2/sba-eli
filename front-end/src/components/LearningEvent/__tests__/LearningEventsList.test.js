import React from "react";
import { List } from "semantic-ui-react";
import { LearningEventsList } from "../LearningEventsList";
import { shallow, mount } from "enzyme";

describe("LearningEventsList", () => {
  it("should render a <List>", () => {
    const wrapper = shallow(<LearningEventsList dispatch={() => {}} />);

    expect(wrapper.find(List).length).toEqual(1);
  });

  it("should render at least one <List.Item>", () => {
    const wrapper = mount(<LearningEventsList dispatch={() => {}} course_id="1" module_id="1" />);

    expect(wrapper.find("li").length).toBeGreaterThanOrEqual(1);
  });
});
