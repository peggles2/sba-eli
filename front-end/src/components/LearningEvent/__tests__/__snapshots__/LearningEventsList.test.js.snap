// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningEventsList should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningEventsList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <li>
        No data
      </li>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No data",
      },
      "ref": null,
      "rendered": "No data",
      "type": "li",
    },
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <li>
          No data
        </li>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No data",
        },
        "ref": null,
        "rendered": "No data",
        "type": "li",
      },
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
